filter/generic/generalizer/json.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package generalizer
+
+import (
+	"encoding/json"
+	"reflect"
+	"sync"
+)
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/common/logger"
+	"dubbo.apache.org/dubbo-go/v3/protocol/dubbo/hessian2"
+	hessian "github.com/apache/dubbo-go-hessian2"
+	perrors "github.com/pkg/errors"
+)
+
+var (
+	jsonGeneralizer     Generalizer
+	jsonGeneralizerOnce sync.Once
+)
+
+func GetJsonrpcGeneralizer() Generalizer {
+	jsonGeneralizerOnce.Do(func() {
+		jsonGeneralizer = &JsonGeneralizer{}
+	})
+	return jsonGeneralizer
+}
+
+type JsonGeneralizer struct{}
+
+func (JsonGeneralizer) Generalize(obj interface{}) (interface{}, error) {
+	newObj, ok := obj.(hessian.Object)
+	if !ok {
+		return nil, perrors.Errorf("unexpected type of obj(=%T), wanted is hessian object", obj)
+	}
+
+	jsonbytes, err := json.Marshal(newObj)
+	if err != nil {
+		return nil, err
+	}
+
+	return string(jsonbytes), nil
+}
+
+func (JsonGeneralizer) Realize(obj interface{}, typ reflect.Type) (interface{}, error) {
+	jsonbytes, ok := obj.(string)
+	if !ok {
+		return nil, perrors.Errorf("unexpected type of obj(=%T), wanted is string", obj)
+	}
+
+	// typ represents a struct instead of a pointer
+	for typ.Kind() == reflect.Ptr {
```

Comment:
`typ` could be a pointer, this for loop may cause some problems(I'm not sure). You should test this branch in unit tests.